src/main/java/seedu/address/logic/commands/EditCommand.java

```diff
@@ -114,19 +117,19 @@ private static Person createEditedPerson(Person personToEdit, EditPersonDescript
             CommandException {
         assert personToEdit != null;
 
-        Name updatedName = editChecker(editPersonDescriptor.getName(), personToEdit.getName());
-        Faculty updatedFaculty = editChecker(editPersonDescriptor.getFaculty(), personToEdit.getFaculty());
-        Phone updatedPhone = editChecker(editPersonDescriptor.getPhone(), personToEdit.getPhone());
-        Email updatedEmail = editChecker(editPersonDescriptor.getEmail(), personToEdit.getEmail());
-        Address updatedAddress = editChecker(editPersonDescriptor.getAddress(), personToEdit.getAddress());
-        MatriculationNumber updatedMatriculationNumber = editChecker(
-                editPersonDescriptor.getMatriculationNumber(), personToEdit.getMatriculationNumber());
-        CovidStatus updatedCovidStatus = editChecker(editPersonDescriptor.getCovidStatus(),
-                personToEdit.getStatus());
-
+        Name updatedName = editPersonDescriptor.getName().orElse(personToEdit.getName());
+        Block updatedBlock = editPersonDescriptor.getBlock().orElse(personToEdit.getBlock());
+        Faculty updatedFaculty = editPersonDescriptor.getFaculty().orElse(personToEdit.getFaculty());
+        Phone updatedPhone = editPersonDescriptor.getPhone().orElse(personToEdit.getPhone());
+        Email updatedEmail = editPersonDescriptor.getEmail().orElse(personToEdit.getEmail());
+        Address updatedAddress = editPersonDescriptor.getAddress().orElse(personToEdit.getAddress());
+        MatriculationNumber updatedMatriculationNumber = editPersonDescriptor.getMatriculationNumber()
+                .orElse(personToEdit.getMatriculationNumber());
+        CovidStatus updatedCovidStatus = editPersonDescriptor.getCovidStatus()
+                .orElse(personToEdit.getStatus());
```

Comment:
I think you overwritten Kevin's changes for the updated fields, which might cause the test case to fail.

`Name updatedName = editChecker(editPersonDescriptor.getName(), personToEdit.getName());`

Reply:
Thanks for noticing! This must have happened during an incorrectly resolved merged conflict 👍